src/main/java/seedu/address/model/ModelManager.java

```diff
@@ -213,16 +248,35 @@ public void updateFilteredPersonList(Predicate<Person> predicate) {
         filteredPersons.setPredicate(predicate);
     }
 
+    //=========== ModuleList ================================================================================
     @Override
-    public boolean isDefaultPresent(Mod mod) {
+    public boolean isDefaultGroupOfModPresent(Mod mod) {
         requireNonNull(mod);
-        return mod.getDefaultGroup() != null;
+        Mod modInList = this.getMod(mod).get();
+        return modInList.getDefaultGroup() != null;
     }
 
     @Override
-    public boolean doesModExist(Mod mod) {
+    public boolean doesModExistInList(Mod mod) {
         requireNonNull(mod);
-        return new UniqueModuleList().contains(mod.value);
+        return moduleList.contains(mod);
+    }
+
+    @Override
+    public String getDefaultGroupOfMod(Mod mod) {
+        Mod modInList = this.getMod(mod).get();
+        return modInList.getDefaultGroup();
+
+    }
+
+    @Override
+    public void addMod(Mod mod) {
+        moduleList.add(mod);
```

Comment:
Might want to include `RequireNonNull(mod)` here, we wouldnt want nulls to be included in our module list

Reply:
Similarly for the other methods in this part too